crates/store/re_types_core/src/tuid.rs

```diff
@@ -40,92 +41,37 @@ impl Loggable for Tuid {
 
     #[inline]
     fn to_arrow<'a>(
-        data: impl IntoIterator<Item = impl Into<std::borrow::Cow<'a, Self>>>,
+        iter: impl IntoIterator<Item = impl Into<std::borrow::Cow<'a, Self>>>,
     ) -> crate::SerializationResult<ArrayRef>
     where
         Self: 'a,
     {
-        let (time_ns_values, inc_values): (Vec<_>, Vec<_>) = data
-            .into_iter()
-            .map(Into::into)
-            .map(|tuid| (tuid.nanoseconds_since_epoch(), tuid.inc()))
-            .unzip();
-
-        let values: Vec<ArrayRef> = vec![
-            Arc::new(UInt64Array::from(time_ns_values)),
-            Arc::new(UInt64Array::from(inc_values)),
-        ];
-        let validity = None;
-
-        Ok(Arc::new(StructArray::new(
-            Fields::from(vec![
-                Field::new("time_ns", DataType::UInt64, false),
-                Field::new("inc", DataType::UInt64, false),
-            ]),
-            values,
-            validity,
-        )))
+        let iter = iter.into_iter();
+
+        let mut builder = FixedSizeBinaryBuilder::with_capacity(iter.size_hint().0, BYTE_WIDTH);
+        for tuid in iter {
+            #[allow(clippy::unwrap_used)] // Can't fail because `BYTE_WIDTH` is correct.
+            builder.append_value(tuid.into().as_bytes()).unwrap();
+        }
+
+        Ok(Arc::new(builder.finish()))
```

Comment:
That would probably make it easier on both arrow-rs and rustc:
```suggestion
        FixedSizeBinary::try_from_iter(iter.map(|tuid| tuid.into().as_bytes()).map(Arc::new)

Reply:
Except `FixedSizeBinary::try_from_iter` fails on zero-length input, so would need an `if`-statement